File: curso_java/15_excepciones/ejemplos/Dividir.java
/*
 * try/catch basico: manejar ArithmeticException.
 */
public class Dividir {
    static int dividir(int a, int b) {
        return a / b;              // si b es 0, lanza ArithmeticException
    }

    public static void main(String[] args) {
        try {
            System.out.println("10 / 2 = " + dividir(10, 2));   // 5
            System.out.println("10 / 0 = " + dividir(10, 0));   // lanza
            System.out.println("no se imprime");
        } catch (ArithmeticException e) {
            System.out.println("Error atrapado: " + e.getMessage());
        }
        System.out.println("El programa continua normalmente");
    }
}

File: curso_c_cpp/cpp/12_excepciones/ejemplos/dividir.cpp
/*
 * throw / try / catch basico: division segura.
 */
#include <iostream>
#include <stdexcept>
using namespace std;

double dividir(double a, double b) {
    if (b == 0) {
        throw runtime_error("division entre cero");
    }
    return a / b;
}

int main() {
    try {
        cout << "10 / 2 = " << dividir(10, 2) << endl;   // 5
        cout << "10 / 0 = " << dividir(10, 0) << endl;   // lanza
        cout << "no se imprime\n";
    } catch (const runtime_error &e) {
        cout << "Error atrapado: " << e.what() << endl;
    }
    cout << "El programa continua normalmente\n";
    return 0;
}
